packages/theme-language-server-common/src/server/startServer.ts

```diff
@@ -203,6 +210,26 @@ export function startServer(
     return blocks.map(([uri]) => path.basename(uri, '.liquid'));
   }
 
+  // TODO: Fix this; seems very costly
+  async function getLiquidFiles(uri: string) {
+    const rootUri = await findThemeRootURI(uri);
+
+    const blockFiles = await fs.readDirectory(path.join(rootUri, 'blocks'));
+    const sectionFiles = await fs.readDirectory(path.join(rootUri, 'sections'));
+    const snippetFiles = await fs.readDirectory(path.join(rootUri, 'snippets'));
+
+    const sources = [];
+    for(let [uri, _] of [...blockFiles, ...sectionFiles, ...snippetFiles]) {
+      const doc = documentManager.get(uri);
+      if (!doc || doc.type !== SourceCodeType.LiquidHtml) {
+        continue;
+      }
+      sources.push(doc);
+    }
+
+    return sources;
+  }
```

Comment:
Should we keep a memoized ast for each file and invalidate it on file change? Reading every file when someone looks up references to the variable seems expensive.